packages/calypso-products/src/is-marketplace-product.ts

```diff
@@ -0,0 +1,10 @@
+/**
+ * Internal dependencies
+ */
+import { YOAST_SEO } from './constants';
+import { formatProduct } from './format-product';
+
+export function isMarketplaceProduct( product : any ): boolean {
+	product = formatProduct( product );
+	return product.productSlug === YOAST_SEO;
```

Comment:
`formatProduct` explicitly transforms `productSlug` into `product_slug`, so you probably don't need to format the product here if you're expecting the camelCase version.